Revit_Core_Engine/Query/ViewSolid.cs

```diff
@@ -87,7 +86,45 @@ private static Solid CropBoxSolid(this View view)
                 return GeometryCreationUtilities.CreateExtrusionGeometry(loops, range.Direction, range.Length);
             }
             else
+            {
+                if (createIfNoCropBox)
+                    return NoCropBoxSolid(view);
+
                 return null;
+            }
+        }
+
+        /***************************************************/
+
+        private static Solid NoCropBoxSolid(this View view)
+        {
+            if (view is ViewPlan viewplan)
+            {
+                Output<double, double> range = viewplan.PlanViewRange();
+                BoundingBoxXYZ viewBBox = new BoundingBoxXYZ
+                {
+                    Min = new XYZ(-1e6, -1e6, range.Item1),
+                    Max = new XYZ(1e6, 1e6, range.Item2)
```

Comment:
Could we please introduce a local variable `private static double m_DefaultHorizontalExtents = 1e+6;`?

Reply:
I used existing `m_DefaultExtents` that is in used inside `ViewDepthLine` method. I think we should centralize this value.

Reply:
Please ignore previous comment, I added `m_DefaultHorizontalExtents` as suggested 👍